pkg/reconciler/managed/reconciler.go

```diff
@@ -525,6 +525,43 @@ func (r *Reconciler) Reconcile(_ context.Context, req reconcile.Request) (reconc
 		"external-name", meta.GetExternalName(managed),
 	)
 
+	// If managed resource has a deletion timestamp and and a deletion policy of
+	// Orphan, we do not need to observe the external resource before attempting
+	// to unpublish connection details and remove finalizer.
+	if meta.WasDeleted(managed) && managed.GetDeletionPolicy() == xpv1.DeletionOrphan {
+		log = log.WithValues("deletion-timestamp", managed.GetDeletionTimestamp())
+		managed.SetConditions(xpv1.Deleting())
+
+		// Empty ConnectionDetails are passed to UnpublishConnection because we
+		// have not retrieved them from the external resource.
```

Comment:
```suggestion
		// Empty ConnectionDetails are passed to UnpublishConnection because we
		// have not retrieved them from the external resource. In practice we currently
		// only write connection details to a Secret, and we rely on garbage collection
		// to delete the entire secret, regardless of the supplied connection details.
```